presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1338,6 +1342,14 @@ else if (criteria instanceof JoinOn) {
                     analysis.addCoercion(expression, BOOLEAN, false);
                 }
 
+                Expression orExpression = findORExpression(expression);
+
+                if (orExpression != null) {
+                    NodeLocation orLocation = orExpression.getLocation().get();
+                    String warningMessage = format("line %s:%s: %s", orLocation.getLineNumber(), orLocation.getColumnNumber(), "JOIN conditions with an OR can cause performance issues as it may lead to a cross join with filter");
```

Comment:
Moving this string format logic to a helper would encourage re-use. There are other places in this analyzer where it could be very nice to include line/column numbers in performance warnings.

Reply:
Yeah - we can open a separate issue for refactoring this analyzer warning/error handling.